### ISSUE TITLE
Новые категории тегов

### ISSUE THREAD

**Issue by 7nik** [stalkerg/anime-pictures#32]
1. Компания / студия - отнести сюда производителей чего бы то ни было. Хотя глянув теги типа иной копирайт, у меня встал вопрос о толковании копирайта-продукта.
2. Примечание - отнести сюда обязательные, по возможности, теги: оф. арт, скан, косплейные теги, обрезанное, отражённый, совместная работа, заимствованный персонаж, jpeg артефакты, rendered, векторная графика, otoko no ko, обложка.

**Comment by S-kidysh** [comment]
Было предложение сделать гораздо больше категорий тегов, чтобы было удобнее проверять/смотреть/добавлять теги.
Например, категория "волосы", где будут лежать длина, цвет, причёска или "одежда": куртки, чулки, майки и т.д.

**Comment by 7nik** [comment]
Ну я лично не вижу особой разницы между объектом и описание, тем более что там много чего попутано в сторону описания.

**Comment by S-kidysh** [comment]
Согласен, тоже не вижу особого смысла в них, это как раз одна из причин почему нужны более подробные категории.

**Comment by 7nik** [comment]
Основанная проблема в том, что кто-то должен все эти теги поперераспределять.
Вспомнил, я ещё хотел категорию группа, но это не просто категория, а скорее тип. Это теги, которые должны быть в базе, но их нельзя добавить на картинку. Их задача группировать теги. Например жесты:  v, mmm, fox shadow puppet, ok sign, finger gun.

**Comment by S-kidysh** [comment]
Почему они должны быть в базе, почему это должен быть тип? просто это очередная отдельная категория жесты. похожая категория - эмоции, где все смайлики, плачи, радости.
Не обязательно абсолютно все теги сразу пихать в категории, это будет постепенная работа, те что не определены будут пока лежать в своих объектах/описаниях.
В чём проблема? Давайте я этим займусь

**Comment by 7nik** [comment]
Можно по-находить десятки групп тегов, которые нужно бы как-то сгруппировать, но тогда имхо список категорий будет слишком большой. Да и возможно будет неплохо иметь возможность вкладывать одну группу в другую.

**Comment by iDolmatov** [comment]
Желательно иметь базовую категорию тега, которая одна на тег и которая отображается на теги.
И категории-фильтры, которые помогут найти тег на вкладке тегов.
Например: Синее небо - базовый тег "небо", но для него описание "синее небо". При этом синее небо остаётся в каком-то виде объект.
Можно в фильтрах галочку "базовые тег", тогда второй тег будет скрыт.
___
Раньше писал в личку

> Всего где-то 48 тысяч тегов. Понятно, что для перебора всех может потребоваться много времени. Но описательных тегов и объектов можно выделить 1-3 тыс., что реально разобрать, при наличии свободного времени. за не очень продолжительное время - думаю, от 1-3 дней, до 1-2 месяцев - по загруженности человека. 
> -- 
> Но, можно и иначе поступить, реализовать при установке тега модератором какую-нибудь удобную подсказку, сразу видимую, где указан тип тега. В базе у тега на сайте есть графа, где перечислены его свойства (обувь, волосы, одежда). Если модератор видит, что тег без категории или не ещё можно категорию указать, то он переходит к тегу и может написать\убрать категорию. Для модератора это будет не обязательно, но он уже знает что за тег ставит и может потратить 5-10 секунд на категоризацию тега. Таким образом база будет наполняться без выделения специального времени на конкретную задачу. 
> ==== 
> Реализовать можно как-то так: 
> http://moemesto.ru/Alex_dolmatov/file/15028363/%D0%95%D1%83%D0%BF%D1%8B.jpg

**Comment by 7nik** [comment]
Так и не понял, что ты имел в виду, равно как и цель твоего предложения.

**Comment by iDolmatov** [comment]
Что сложного? Обычный фильтр. 
Цитата это уже многоуровневый вариант фильтровки. Это просто что раньше предлагал.
Пример базы:
Корневой тег с которым связываются другие теги для поиска==тип тега||дочерние теги==тип дочерних тегов
Волосы==объект||розовые волосы, каштановые волосы == цвет волос||длинные волосы, короткие волосы == длинна волос.
Мимика = эмоция||улыбка, ;d == выражения лица||радость, грусть == общие эмоции
Первая колонка объект, последующие описания.
Многоуровневый фильтр:
Тип тега, Подтип тега, результат
1. Объект = Волосы, кольца - можно выбрать все или отключить часть ||мимика, жесты
2. Подтип = если выбран 1 с "объектом", то можно выбрать описание для подтипа
3. Цвет волос
Результат: розовые волосы, каштановые волосы.

Возможно, стоит как-то графически рисовать для понимания. Но пока есть время только так.

**Comment by 7nik** [comment]
На мной взгляд через чур сложно. Система должна быть такой, чтобы свежеиспечённый пользователь мог пользоваться ею практически без обучения.
И что вообще это должно искать? картинки или теги?

**Comment by iDolmatov** [comment]
Это фильтр для тегов, только расширенный.
По сути всего до трёх уровней. Желательно до двух.
Тип тега = объект, описание
Подтип, если выбран тип. Это стиль тега = описание объекта или для описания описания.

Выбрал объектные типы.
В них выбрал вариант "причёски" или "цвет волос".

Выбрал описание:
Виды ландшафта "поле", "городской пейзаж".

**Comment by 7nik** [comment]
Вроде как теперь понял.
Имхо, моё предложение о группах тегов удобнее. Находишь, например, группу волосы, в неё вложены группы цвет волос, длина волос, причёска и просто тег развевающиеся волосы, открываешь нужную группу и смотришь теги. При этом ты можешь перемещаться как вниз к конкретным тегам, так и вверх к более общи группам.

**Comment by S-kidysh** [comment]
> имхо список категорий будет слишком большой

не думаю, их по пальцам можно пересчитать

> возможность вкладывать одну группу в другую

это лишнее, достаточно будет иерархии в том виде, который сейчас: категория->тег
в большинстве случаев в категории "волосы" будет лежать 2 тега, например: синие волосы, длинные волосы, поэтому создавать подкатегории длина волос или цвет волос - пустая трата места.

**В итоге**
всё уже реализовано, работа не пыльная, достаточно просто добавить десяток-другой "типов тегов"(категорий): "волосы", "глаза", "одежда" и т.д. и быстренько разбросать в них большинство тегов, вот и всё.

**Comment by S-kidysh** [comment]
волосы (длина, цвет, причёска..)
лицо (цвет глаз, мимика...)
одежда (бикини, платья, майки...)
окружение (фон, мебель, поля...)
растения (цветы, деревья...)
действия (жесты, удары, обнимашки...)
эмоции (радость, смайлики...)
украшения (укр. для волос, кулоны, заколки...)
текст (надписи, подписи, выноски...)

что-то мб ещё забыл? если в какой-то категории будет очень много тегов по сравнению с остальными, можно разделить на несколько более конкретных.

**Comment by 7nik** [comment]
Мне всё равно больше нравится мой вариант с группами.

Надо ещё чьё-то мнение послушать.

**Comment by S-kidysh** [comment]
одно другому не мешает, я лишь привёл группы первой степени, можно продолжать и дальше иерархию, я не спорю, мне в любом виде бы понравились изменения
